docs/docs/advanced/deploy-eliza-to-tee-with-oasis-rofl.md

```diff
@@ -0,0 +1,257 @@
+
+    ```bash
+    oasis rofl create --network testnet --account <account-name>
+    ```
+
+### Build and Deploy the ROFL app
+
+1. **Configure the ROFL app**
+
+    Update the `rofl-compose.yaml` file with your configuration:
+
+    ```yaml
+    services:
+      eliza:
+        restart: always
+        image: ghcr.io/elizaos/eliza:latest
+        build:
+          context: .
+          dockerfile: Dockerfile
+        platform: linux/amd64
+        environment:
+          - CHARACTERS=${CHARACTERS}
+          - SERVER_PORT=${SERVER_PORT}
+          # Add other required environment variables
+    ```
+
+2. **Encrypt Secrets**
+
+    Encrypt all required environment variables. You can use any character file of your choice - here we're using the staff engineer character as an example. Refer to the Secret Management section above for a comprehensive list of secrets you might want to set:
+
+    ```bash
+    # Set the character file path (you can use any character file of your choice)
+    echo -n "characters/staff-engineer.character.json" | oasis rofl secret set CHARACTERS -
+    echo -n "3000" | oasis rofl secret set SERVER_PORT -
+    # Add other required secrets
+    ```
+
+    **WARNING:** Secrets are end-to-end encrypted with your account key and an ephemeral key of the chain where it will be deployed to (Testnet or Mainnet). Migrating encrypted secrets from one deployment to another is not possible.
+
+3. **Configure ROFL Resources**
+
+    Adjust the resource allocation in your `rofl.yaml` file to match your application's requirements. This includes memory, CPU, and storage settings:
+
+    ```yaml
+    resources:
+      memory: 16384  # Memory in MB (16GB in this example)
+      cpus: 1        # Number of CPU cores
+      storage:
+        kind: disk-persistent
+        size: 10240  # Storage size in MB (10GB in this example)
+    ```
+
+    Adjust these values based on your specific needs:
+    - `memory`: Set the amount of RAM your application needs (in MB)
+    - `cpus`: Specify the number of CPU cores to allocate
+    - `storage`: Configure persistent storage with the required size
+
+4. **Build the ROFL Application**
+
+    For MacOS users, use the following Docker image:
+
+    ```bash
+    docker run --platform linux/amd64 --volume .:/src --rm -it ghcr.io/oasisprotocol/rofl-dev:main oasis rofl build
+    ```
+
+5. **Deploy the ROFL Application**
+
+    The recommended way to deploy your ROFL application is using the Oasis CLI:
+
+    ```bash
+    oasis rofl deploy
+    ```
+
+    This command will:
+    1. Check if you have an existing machine available
+    2. If no machine exists, it will:
+       - Present a list of available providers
+       - Allow you to select a provider and plan for a TDX-capable machine
+       - Handle the payment process for renting the machine
+    3. Deploy your ROFL application to the selected machine
+
+    You can deploy to:
+    - Your own nodes (if you're the provider)
+    - Oasis-sponsored nodes
```

Comment:
Sponsored is maybe too generous. For Mainnet, people will still need to pay ROSE (and/or other currencies) for renting the machine.

```suggestion
    - Nodes managed by the Oasis Protocol Foundation
```